File: scrape_accessories.js
const https = require('https');

const urls = [
    "https://a-key-gmbh.com/OBDSTAR-Zubehoer",
    "https://a-key-gmbh.com/XHORSE-Zubehoer",
    "https://a-key-gmbh.com/Zed-FULL-Zubehoer",
    "https://a-key-gmbh.com/Autel-Zubehoer"
];

function fetchHTML(url) {
    return new Promise((resolve, reject) => {
        https.get(url, { headers: { 'User-Agent': 'Mozilla/5.0' } }, (res) => {
            let data = '';
            res.on('data', chunk => data += chunk);
            res.on('end', () => resolve(data));
        }).on('error', reject);
    });
}

async function scrape() {
    const results = [];
    for (let url of urls) {
        const brand = url.split('/').pop().split('-')[0];
        const html = await fetchHTML(url);
        
        // Very basic regex scraping
        const regex = /class="product-title"[^>]*>\s*<a href="([^"]+)"[^>]*>(.*?)<\/a>/gs;
        let match;
        while ((match = regex.exec(html)) !== null) {
            results.push({
                brand: brand,
                link: match[1],
                title: match[2].trim()
            });
        }
    }
    console.log(JSON.stringify(results, null, 2));
}

scrape();

File: scrape_accessories.py
import urllib.request
from bs4 import BeautifulSoup
import json
import csv

urls = [
    "https://a-key-gmbh.com/OBDSTAR-Zubehoer",
    "https://a-key-gmbh.com/XHORSE-Zubehoer",
    "https://a-key-gmbh.com/Zed-FULL-Zubehoer",
    "https://a-key-gmbh.com/Autel-Zubehoer"
]

results = []

for url in urls:
    req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
    html = urllib.request.urlopen(req).read()
    soup = BeautifulSoup(html, 'html.parser')
    
    # Items are usually inside .product-wrapper or .product-info
    items = soup.find_all('div', class_='product-wrapper')
    brand = url.split('/')[-1].split('-')[0]
    
    for item in items:
        title_elem = item.find('a', class_='text-clamp-2')
        if not title_elem:
            continue
        title = title_elem.text.strip()
        link = title_elem.get('href')
        
        price_elem = item.find('div', class_='product-price')
        price = price_elem.text.strip() if price_elem else ""
        
        img_elem = item.find('img')
        img = img_elem.get('src') if img_elem else ""
        
        results.append({
            'brand': brand,
            'title': title,
            'link': link,
            'price': price,
            'image': img
        })

print(json.dumps(results, indent=2))
